base/experiments/ad9910_ram.py

```diff
@@ -0,0 +1,274 @@
+from artiq.experiment import *
+from artiq.coredevice.ad9910 import *
+import numpy as np
+from enum import Enum
+
+
+__all__ = ["AD9910RAM"]
+
+
+class RAMType(Enum):
+    """The type of data in the RAM.
+
+    Enums:
+        FREQ: Data are treated as frequencies (Hz).
+            The exact frequencies will be played-back.
+        PHASE: Data are treated as phases (turns)
+            Phase modulation will be performed.
+        AMP: Data are treated as amplitudes.
+            Amplitude modulation will be performed.
+        Polar: Data are treated as amplitudes & phase.
+            Polar modulation will be performed.
+    """
+    FREQ = 1
+    PHASE = 2
+    AMP = 3
+    POLAR = 4
+
+
+class RAMProfile:
+    # From AD9910 datasheet:
+    # "The AD9910 makes use of a 1024 × 32-bit RAM."
+    # Each entry occupies 32 bits.
+    RAM_SIZE = 1024
+
+    def __init__(self, dds, data, ramp_interval, ram_type, ram_mode,
+                 base_frequency=0, base_phase=0, base_amplitude=1.0):
+        """Generate a RAM profile to a DDS channel.
+
+            RAM profile is an operation mode that feeds RAM data into the DDS
+            as DDS parameter(s) (frequency/phase/amplitude). At regular time
+            interval, the DDS fetches a new RAM data and updates the selected
+            parameter. See the arguments to tune to interval, DDS parameter
+            and data selection. (See args ramp_interval, ram_type, ram_mode)
+
+            Unlike single-tone profiles, RAM profiles only specifies a subset
+            of parameters. The remaining parameters are specified through the
+            FTW/POW/ASF registers instead (See args base_frequency, base_phase,
+            base_amplitude). Note that unnecessary paramters will be ignored.
+
+            For example, a frequency RAM profile feeds RAM as frequencies.
+            Phase and amplitude are specified through POW and ASF registers
+            respectively. These registers are controlled by the base_phase and
+            base_amplitude args.
+
+            Reference: AD9910 datasheet (Rev. E), Theory of Operation, Mode
+            Priority, Table 5: Data Source Priority
+
+            Note: There are different data latencies between different RAM
+            types. Amplitude takes 36 fewer sysclk cycles to reach the DDS
+            output. Refer to AD9910 datasheet (Rev. E) p.6, Data Latency
+
+        Args:
+            dds (AD9910): The DDS that will playback the RAM profile.
```

Comment:
Can you edit the docstring style to match other code? See https://github.com/Jayich-Lab/pydux/blob/main/DOCSTRING_STYLE.md as an example.